frontend/src/hooks/useEvent.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useCallback, DependencyList } from 'react'
+
+/**
+ * Specifies what a function should look like that handles an event.
+ * This saves us specifying the `detail` parameter everywhere and makes some logic simplier
+ */
+type EventHandler<T> = (arg: {detail: T}) => void
+
+/**
+ * Mapping of events to what data the event accepts.
+ * If making a custom event just add it here first
+ */
+export interface Events {
+  'editTransition': {id: number},
+  'editComment': {id: number, x: number, y: number},
+  'editStateName': {id: number},
+  'editStateLabel': {id: number},
+}
+
+interface EventOptions {
+  target: Document,
+  options?: any // boolean | AddEventListenerOptions
```

Comment:
I don't really know what options is meant to be, I tried to find a place it is used in the code but couldn't yet.
I tried using the type specified by typescript for `addEventListener` but the types it uses aren't exported so I can't use them

Reply:
Never mind found a few uses, will clean it up